fennel.lua

```diff
@@ -2814,7 +2817,7 @@ that argument name begins with ?."
                   (do (assert (not (. pattern (+ k 2)))
                               "expected rest argument in final position")
                       (table.insert bindings (. pattern (+ k 1)))
-                      (table.insert bindings [`(select ,k ((or unpack table.unpack)
+                      (table.insert bindings [`(select ,k ((or _G.unpack table.unpack)
```

Comment:
I don't think `_G.unpack` is necessary here, since we're explicitly exposing `unpack` in the compiler scope to avoid having to account for it in macros and such.

Also, I think we want to account for scenarios where an environment is set without a `_G`, which is why we've avoided using it directly in the rest of the compiler?

Reply:
The funny thing is without this you can't use `match` when you have strict globals mode turned on in 5.3+, because `(or unpack table.unpack)` is included in the macroexpansion.

Reply:
Huh, bit of a catch-22 there. On the one hand it'll break with strict globals, on the other hand it'll break in runtime if run in an environment that doesn't include `_G` itself. I suppose the latter is pretty rare, but I can't help but wish we could have it both ways!